### ISSUE TITLE
Android Manifest is not updated

### ISSUE THREAD

**Issue by dansiegel** [dansiegel/Mobile.BuildTools#186]
# Description

When building the Android Manifest should be updated with a new build number and any tokens should be replaced from the manifest.json or secrets.json. This does not occur

## Expected Behavior

Build number and tokens should be updated

## Actual Behavior

Build number and tokens are not updated

## Reproduction App

E2E / sample apps

**Comment by dansiegel** [comment]
Note: We probably need to hook into the [AfterGenerateAndroidManifest](https://docs.microsoft.com/en-us/xamarin/android/deploy-test/building-apps/build-properties#aftergenerateandroidmanifest) task that was added in Xamarin Android 9.2

**Comment by dansiegel** [comment]
moved to the 2.1 milestone

**Comment by dansiegel** [comment]
looking more at this we will want to hook into the `_GenerateJavaStubs` Target. 

The Mobile.BuildTools will need to update the `_AndroidManifestAbs` property so that it points at the version of the input manifest that the Mobile.BuildTools creates.

https://github.com/xamarin/xamarin-android/blob/4ed6a222ed07551734b28525dd269d50ac107ea8/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1401-L1405

we'll likely want to ensure that the Mobile.BuildTools adds itself in the DependsOn pipeline using `_GenerateJavaStubsDependsOnTargets`

```xml
<PropertyGroup>
  < _GenerateJavaStubsDependsOnTargets>$(_GenerateJavaStubsDependsOnTargets);MobileBuildToolsTarget</_GenerateJavaStubsDependsOnTargets>
</PropertyGroup>
```

**Comment by biapar** [comment]
The same issue for me. Xamarin Forms 5.0

**Comment by dansiegel** [comment]
Fixed by #238 & #253
